app/services/secure_headers_whitelister.rb

```diff
@@ -4,7 +4,8 @@ def self.extract_domain(url)
   end
 
   def run
-    whitelisted_domains = domains(acs_urls(SERVICE_PROVIDERS.values))
+    return unless ActiveRecord::Base.connection.table_exists? 'service_providers'
```

Comment:
I mentioned this in Slack, but I think it may be worth discussing: What if we removed the `SecureHeadersWhitelister` entirely? Or at least remove it from the initializers?

I think what we can do instead is set the secure headers in the IDP controller, and override the `form_action` values there, rather than set a pre-configured list of all the headers here.

The advantage:
1. no DB calls in an initializer
2. less race condition prone: the code would read the values as they exist in the DB (this is still susceptible to the 2 boxes behind a load balancer problem, where one box may get the API call to update the database, but the other box may not know to update its saved configuration)
3. we can look up the `acs_url` for the particular SP in the controller, rather than providing a list of all SPs, so we would expose less